### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by jowkame**

### CODE REVIEW

File: LibraryModule.rb [hunk 1]
```diff
@@ -7,16 +7,19 @@
 
 module LibraryModule
```
Comment (jowkame) [inline LibraryModule.rb]: to @yamnich
I've updated LibraryModule with several improvements.

File: LibraryModule.rb [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
+
+    def fetch_orders_data(field)
+      @orders.group_by(&:"#{field}").map {|key, value| [key, value.length]}.sort_by {|key, value| value}.reverse
+    end
+
+    def most_active_reader
+      fetch_orders_data('reader').first[0].name unless @orders.nil?
+    end
+
+    def most_popular_book()
```
Comment (yamnich) [inline LibraryModule.rb]: remove () please

File: LibraryModule.rb [hunk 3]
```diff
@@ -1,8 +1,51 @@
...
+        file.write(data_hash.to_yaml)
+      end
+    end
+
+    def fetch_orders_data(field)
+      @orders.group_by(&:"#{field}").map {|key, value| [key, value.length]}.sort_by {|key, value| value}.reverse
+    end
+
+    def most_active_reader
+      fetch_orders_data('reader').first[0].name unless @orders.nil?
```
Comment (yamnich) [inline LibraryModule.rb]: do we need to use unless @orders.nil? here ?

File: main.rb [hunk 4]
```diff
@@ -99,9 +99,13 @@
...
+another_library.open('lib.yml')
+
+puts 'most active reader'
+puts another_library.most_active_reader, ''
+
+puts 'most popular book:'
+puts another_library.most_popular_book, ''
+
+puts 'books ordering statistics:'
+puts another_library.books_ordering_statistics&.map{|book, count| [book.title, count]}
```
Comment (yamnich) [inline main.rb]: this should return count of unique readers who orders one of 3 popular books

**Pull request merged**